src/pyenphase/auth.py

```diff
@@ -93,16 +96,27 @@ async def setup(self, client: httpx.AsyncClient | None = None) -> None:
                 "Unable to obtain token for Envoy authentication."
             )
 
+        # Verify the token and obtain cookie with session ID necessary for some API calls
+        req = await client.get(
+            f"https://{host}/auth/check_jwt",
+            headers={"Authorization": f"Bearer {self.token}"},
+        )
+
+        if req.status_code != 200:
+            raise EnvoyAuthenticationError(
+                "Unable to verify token for Envoy authentication."
+            )
```

Comment:
If it returns more information it would be nice to include the status code/text so its clear whats wrong.

Reply:
No information, just the default redirection to the login page